modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1054 @@
+##' call rp_control (from RpTools Python package) and store the output in BETY
+##'
+##' @name remote_process
+##' @title remote_process
+##' @export
+##'
+##' @param settings PEcAn settings list containing remotedata tags: raw_mimetype, raw_formatname, source, collection, scale, projection, qc, algorithm, credfile, pro_mimetype, pro_formatname, out_get_data, out_process_data, overwrite
+##'
+##' @examples
+##' \dontrun{
+##' remote_process(settings)
+##' }
+##' @author Ayush Prasad, Istem Fer
+##'
+
+remote_process <- function(settings) {
+  # Information about the date variables used in remote_process:
+  # req_start, req_end : start, end dates requested by the user, the user does not have to be aware about the status of the requested file in the DB
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to rp_control for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+  
+  # The value of remotefile_check_flag denotes the following cases:
+  
+  # When processed file is requested,
+  # 1 - There are no existing raw and processed files of the requested type in the DB
+  # 2 - Requested processed file does not exist, the raw file used to create is it present and matches with the requested daterange
+  # 3 - Requested processed file does not exist, raw file used to create it is present but has to be updated to match with the requested daterange
+  # 4 - Both processed and raw file of the requested type exists, but they have to be updated to match with the requested daterange
+  # 5 - Raw file required for creating the processed file exists with the required daterange and the processed file needs to be updated. Here the new processed file will now contain data for the entire daterange of the existing raw file
+  # 6 - There is a existing processed file of the requested type but the raw file used to create it has been deleted. Here, the raw file will be created again and the processed file will be replaced entirely with the one created from new raw file
+  
+  # When raw file is requested,
+  # 1 - There is no existing raw the requested type in the DB
+  # 2 - existing raw file will be updated
+  
+  RpTools <- reticulate::import("RpTools")
+  
+  input_file             <- NULL
+  stage_get_data         <- NULL
+  stage_process_data     <- NULL
+  raw_merge              <- NULL
+  pro_merge              <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
```

Comment:
I think some of these are set now in the refactored functions, could you clean up the ones that are not needed